test/test_mps.py

```diff
@@ -7781,22 +7781,27 @@ def test_mps_generator(self):
         g_mps.manual_seed(999)
         mps_x = torch.randn(5, device='mps', generator=g_mps)
         g_mps.manual_seed(999)
+        # generate random numbers with offset `0`
         mps_y = torch.randn(5, device='mps', generator=g_mps)
         # seed values were the same, so the random tensor contents should match
         self.assertEqual(mps_x, mps_y)
-        # save generator's state to restore it later
+        # save generator's state (offset = 1) to restore it later
         g_state = g_mps.get_state()
 
-        # generate random numbers without seeding
+        # generate random numbers with offset `1`
         mps_x = torch.randn(5, device='mps', generator=g_mps)
         # in this case, the random results must differ from the last generated random results
         self.assertNotEqual(mps_x, mps_y)
 
+        # mps_x was produced by g_state, we use it as our reference mps_y.
+        mps_y = mps_x
+
         # restore the previously saved state, and the results should match again
         g_mps.set_state(g_state)
         mps_x = torch.randn(5, device='mps', generator=g_mps)
         self.assertEqual(mps_x, mps_y)
```

Comment:
The previous testing approach was flawed. It did not consider Philox's offset at all. The reason that the `assertEqual` passed was that `mps_y` was generated with `offset = 0` and the `g_state` should have had carry `offset = 1` but it didn't. As a result, here when we restored from `g_state`, we implicitly set `offset = 0`, and `mps_x` and `mps_y` coincidentally matched.